### PR TITLE
feat: implemented uniform_

### LINKED ISSUE #26701: fill_diagonal_

### PR DESCRIPTION
**Pull request by MahamedDucale**
# PR Description

## Related Issue

Close #26701

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [ ] Did you run your tests and are your tests passing?
- [x] Did pre-commit not fail on any check?
- [x] Did you follow the steps we provided?

### CODE REVIEW

**Review by hello-fri-end — CHANGES_REQUESTED** [review]
Hey @MahamedDucale , thanks for creating this PR. Would you please add a test for the `uniform_` function as well. Please make sure all the tests are passing locally. Happy to help out if you run into any issues. Thanks :)

**Comment by MahamedDucale** [comment]
@hello-fri-end I added the test. However I still have 1 test failing "array-api-intelligent-tests-pr"

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py [hunk 1]
```diff
@@ -4623,6 +4623,47 @@ def test_paddle_tensor_unbind(
     )
 
 
+# uniform_
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="paddle.to_tensor",
+    method_name="uniform_",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"),
```
Comment (hello-fri-end) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py:4632]:
```suggestion
        available_dtypes=helpers.get_dtypes("valid"),
```

This should always be valid- if the function only supports floats, then that's what valid will return.

**Review by hello-fri-end — CHANGES_REQUESTED** [review]
Hey @MahamedDucale , thanks for making the changes. Just requested another minor change, happy to review again once you're done with the changes :)

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 2]
```diff
@@ -927,21 +926,17 @@ def gather_(self, y, name=None):
...
-            }
+            "2.5.1 and below": (
+                "float32",
+                "float64",
+            )
         },
         "paddle",
     )
-    def tile(self, repeat_times):
-        return paddle_frontend.Tensor(ivy.tile(self._ivy_array, repeats=repeat_times))
```
Comment (hello-fri-end) [inline ivy/functional/frontends/paddle/tensor/tensor.py:947]: Why are we removing this function?

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 3]
```diff
@@ -927,21 +926,17 @@ def gather_(self, y, name=None):
...
-                "cpu": (
-                    "bool",
-                    "int32",
-                    "int64",
-                    "float32",
-                    "float64",
-                    "complex64",
-                    "complex128",
-                )
-            }
```
Comment (hello-fri-end) [inline ivy/functional/frontends/paddle/tensor/tensor.py:942]: Why are we removing these?

**Review by hello-fri-end — CHANGES_REQUESTED** [review]
Hey @MahamedDucale, sorry for the slow response here, I was OOO. Could you please address the comments I have left after which I think this should be good to merge!

**Comment by NiteshK84** [comment]
Closing the PR for the time being, since there has been no activity for more than 2 weeks.

**Pull request closed without merging**